packages/core/src/git/GitTagPurgeListv1.ts

```diff
@@ -0,0 +1,72 @@
+import simpleGit, { SimpleGit } from 'simple-git';
+
+interface Tag {
+  name: string;
+  timestamp: number;
+}
+
+export default class GitTagPurge {
+  private _git: SimpleGit;
+
+  constructor() {
+    this._git = simpleGit();
+  }
+
+  public async listGitTags(pkgs: string[]): Promise<string[]> {
```

Comment:
See whether you can use 
packages/core/src/git/GitTags.ts
and its functions  especially listTagsOnBranch or modify it